bob-apps/bin/bobt.rs

```diff
@@ -0,0 +1,434 @@
+            if i % 10000 == 0 {
+                log::info!("Summary: {}/{}", total_succ, i);
+                self.client.print_summary();
+                self.client.reset_metrics();
+            }
+            let op = Operation::gen(&mut self.rng);
+            let success = match op {
+                Operation::Get => self.get().await,
+                Operation::Put => self.put().await,
+                Operation::Delete => self.delete().await,
+            };
+            if success {
+                total_succ += 1;
+            }
+        }
+    }
+}
+
+struct Client {
+    client: BobApiClient<Channel>,
+    http_client: reqwest::Client,
+    settings: Settings,
+    put_time: Duration,
+    put_count: u64,
+    get_count: u64,
+    get_time: Duration,
+    delete_count: u64,
+    delete_time: Duration,
+}
+
+impl Client {
+    async fn new(settings: Settings) -> Self {
+        Self {
+            client: BobApiClient::connect(settings.uri.clone()).await.unwrap(),
+            http_client: reqwest::ClientBuilder::default().build().unwrap(),
+            settings,
+            put_count: 0,
+            put_time: Duration::ZERO,
+            get_count: 0,
+            get_time: Duration::ZERO,
+            delete_count: 0,
+            delete_time: Duration::ZERO,
+        }
+    }
+
+    fn reset_metrics(&mut self) {
+        self.put_time = Duration::ZERO;
+        self.put_count = 0;
+        self.get_time = Duration::ZERO;
+        self.get_count = 0;
+        self.delete_time = Duration::ZERO;
+        self.delete_count = 0;
+    }
+
+    fn print_summary(&self) {
+        let latency = self.get_time.as_secs_f64() / self.get_count as f64 * 1000.0;
+        log::info!("GET: count: {}, latency: {}", self.get_count, latency);
+        let latency = self.put_time.as_secs_f64() / self.put_count as f64 * 1000.0;
+        log::info!("PUT: count: {}, latency: {}", self.put_count, latency);
+        let latency = self.delete_time.as_secs_f64() / self.delete_count as f64 * 1000.0;
+        log::info!("DELETE: count: {}, latency: {}", self.delete_count, latency);
+    }
+
+    async fn put(&mut self, key: u64, size: usize) -> Result<(), String> {
+        let data = vec![1; size];
+        let mut url = Url::parse(&self.settings.api_uri.to_string()).unwrap();
+        url.set_path(&format!("/data/{}", self.settings.convert_key(key)));
+        let addr = url.to_string();
+        log::info!("Put to {}", addr);
+        let sw = Stopwatch::new();
+        let req = self.http_client
+            .post(addr)
+            .body(data)
+            .header("username", "admin")
+            .header("password", "password")
+            .build()
+            .map_err(|e| e.to_string())?;
+        let res = self.http_client.execute(req).await.map_err(|e| e.to_string())?;
+        if !res.status().is_success() {
+            log::warn!("Put resulted in error code: {:?}", res.status());
+        }
+        log::info!(
```

Comment:
`debug!`

Plus, as I wrote before, it is better to move logging after `put_time` and `put_count` updates, because this operation will affect sw.elapsed() value